### PR TITLE
Implements  booleanOverlap and line_overlap function

### PR DESCRIPTION
**Pull request by armantorkzaban**

### CODE REVIEW

File: lib/src/line_overlap.dart [hunk 1]
```diff
@@ -43,154 +43,148 @@ FeatureCollection<LineString> lineOverlap(
...
   var features = <Feature<LineString>>[];
 
   // Create Spatial Index
   var tree = RBushBase<Feature<LineString>>(
-      getMinY: (Feature<LineString> feature) => bbox(feature).lat1.toDouble(),
-      getMinX: (Feature<LineString> feature) => bbox(feature).lng1.toDouble(),
-      toBBox: (feature) => _toRBBox(feature));
+    getMinX: (Feature<LineString> feature) => bbox(feature).lng1.toDouble(),
+    getMinY: (Feature<LineString> feature) => bbox(feature).lat1.toDouble(),
+    toBBox: (feature) => _toRBBox(feature),
```
Comment (lukas-h) [inline lib/src/line_overlap.dart:55]: Couldn't you just assign the function like this?
`toBBox: _toRBBox,`

**Comment by lukas-h** [comment]
@armantorkzaban we should finish this in the next weeks. This is another "low hanging fruit", which can be finished easily

**Comment by jsiedentop** [comment]
Okay, I need to take a closer look. Some tests fail, and it appears that the implementation is incomplete.

I picked out this `./test/examples/booleans/overlap/true/simple-lines.geojson` example, where two lines partially overlap. Unfortunately, our current implementation failed to detect this overlap, whereas turf.js correctly identifies it.

**Comment by lukas-h** [comment]
@jsiedentop this is high priiority

see this project: https://github.com/opendata-nuernberg/soil_unsealing_potential/tree/main/polygon_analyzer

**Comment by jsiedentop** [comment]
I am closing this because the functions were implemented with #174

**Pull request closed without merging**